### DISCUSSION TITLE
Calling in Solidity function getL1Fee(bytes memory _data) external view returns (uint256);

### DISCUSSION THREAD

**Discussion by shaobaili3** [ethereum-optimism/developers#35] [Application Development] [answered]
In Solidity, if a function changes the on-chain status, does the timing of calling the getL1Fee function affect its outcome? Specifically, would the result be the same if it’s called at the beginning of the function versus just before the return statement?

**Comment by opfocus** [comment]
I don't have a clear understanding of this  @sbvegan

**Comment by joohhnnn** [comment]
Firstly, I personally believe that when using RPC methods like `eth_estimateGas`, in simple terms, a state is instantiated within the RPC method by the node. This state represents the status of all smart contracts after they are executed in the EVM, hence the scenario you mentioned regarding pre-function call and pre-return does not arise.

**Reply by shaobaili3** [reply]
The question is, if this function has already received calldata and consistently returns the gas fee required for the final state of the current transaction (tx), then is this calldata essentially redundant? Because if any input value does not change the final outcome of the current transaction, then the consumed gas fee should be roughly the same (the gas consumed due to the length of the input parameters is negligible and can be ignored).

**Comment by sbvegan** [comment, marked as answer]
> If a function changes the on-chain status, does the timing of calling the getL1Fee function affect its outcome?

Yes it does (across L2 blocks), the L1 base fee is part of the computation, and that changes frequently

> Specifically, would the result be the same if it’s called at the beginning of the function versus just before the return statement?

Within the same L2 block it does not change
